### DISCUSSION TITLE
Getting Error: getaddrinfo ENOTFOUND stagingsupply.htm-mbs.com when trying to deploy to Netlify/Heroku

### DISCUSSION THREAD

**Discussion by nicholastorr** [gatsbyjs/gatsby#34992] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Running Gatsby build everything runs fine but when I try to deploy to a cloud service like Netlify I get the error Error: getaddrinfo ENOTFOUND stagingsupply.htm-mbs.com. This is a valid URL and like I said when I run gatsby build or start on my local computer it fetches the data without a hitch. I've tried reinstalling node, npm, gatsby setting the engines to my version of Gatsby and still getting this error 

### Reproduction Link

https://github.com/nicholastorr/gatsby-d7

### Steps to Reproduce

1. Add repo to Netlify and deploy from main branch

### Expected Result

builds site and is able to host on cloud provider

### Actual Result

![image](https://user-images.githubusercontent.com/52306328/156060527-400568ad-c1d1-47ff-9c88-03310ce0d7df.png)

### Environment

```shell
System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
  Binaries:
    Node: 14.17.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.9.5
  Browsers:
    Chrome: 98.0.4758.102
    Edge: Spartan (44.19041.1266.0), Chromium (98.0.1108.62)
  npmPackages:
    gatsby: 4.6.0 => 4.6.0 
    gatsby-plugin-gatsby-cloud: ^4.8.0 => 4.8.1 
    gatsby-plugin-image: ^2.8.0 => 2.8.1 
    gatsby-plugin-manifest: ^4.8.1 => 4.8.1 
    gatsby-plugin-offline: ^5.8.0 => 5.8.1 
    gatsby-plugin-postcss: ^5.8.0 => 5.8.0 
    gatsby-plugin-purgecss: ^6.1.0 => 6.1.0 
    gatsby-plugin-react-helmet: ^5.8.0 => 5.8.0 
    gatsby-plugin-sharp: ^4.8.0 => 4.8.1 
    gatsby-source-drupal7: 0.0.4 => 0.0.4 
    gatsby-source-filesystem: ^4.8.0 => 4.8.1 
    gatsby-transformer-sharp: ^4.8.0 => 4.8.0 
  npmGlobalPackages:
    gatsby-cli: 4.6.0
```

**Comment by LekoArts** [comment]
Hi! I'd recommend using the official plugin: https://www.gatsbyjs.com/plugins/gatsby-source-drupal/

**Reply by nicholastorr** [reply]
Sorry forgot to mention I am using Drupal 7 and using gatsby-source-drupal7 I believe it is made off of the same plugin

**Reply by nicholastorr** [reply]
anyway I could find some help with this? Gatsby source drupal gives me an error Unexpected token < in JSON at position 0
